source-code/sdk-js/adapter-sveltekit/package.json

```diff
@@ -0,0 +1,4 @@
+{
+	"name": "@inlang/sdk-js-adapter-sveltekit",
```

Comment:
Is this the package-name we want developers to install the adapter?

Reply:
Each adapter should be it's own published npm package. Or else a Next.js project would need to install SvelteKit as a peer dependency.

`@inlang/sdk-js/sveltekit` is not a valid package name. A name can only contain a single `/`.

I just saw that we have a `git-sdk`. Should we then rename this package to `js-sdk` to be consitent?

Reply:
> Or else a Next.js project would need to install SvelteKit as a peer dependency. 

We can unbundle the package in the future if developers start to complain about the size of the package.

Having everything in one package eases deployment, "SEO" because we only need to optimize one package on NPM/Readme, bundles all downloads together (higher chance of adoption / higher ranking on libhunt etc.).  

-------

Which of those SDKs would you be more inclined to adopt? 

```
@inlang/sdk-js             4500 weekly downloads
```

```
@inlang/sdk-js-sveltekit   1100 weekly downloads
@inlang/sdk-js-nextjs       900 weekly downloads
@inlang/sdk-js-astro        500 weekly downloads
@inlang/sdk-js-solid-start  600 weekly downloads
@inlang/sdk-js-react       1200 weekly downloads
```

> I just saw that we have a git-sdk. Should we then rename this package to js-sdk to be consistent? 

I think `sdk-{platform}` is better because the source code would be sorted. The git-sdk is "unrelated" to inlang. For ease of development, it lives in the monorepo for now.

```
website
sdk-js
sdk-flutter
sdk-ios
sdk-android
```

Reply:
Then let's do this for now. We can also change the structure later if we see that it makes sense

Reply:
One big downside that just came to my mind is:
Once we have multiple adapters, we will also have **functions with the same name multiple times**, because all adapters will share the same `API`. If you want to call `changeLanguage`, then your IDE will probably auto import it from `@inlang/sdk-js-svelte` instead of `@inlang/sdk-js-sveltekit`. If you don't pay attention then you will be wondering why your code does not work.

Reply:
>One big downside that just came to my mind is:
Once we have multiple adapters, we will also have functions with the same name multiple times, because all adapters will share the same API. If you want to call changeLanguage, then your IDE will probably auto import it from @inlang/sdk-js-svelte instead of @inlang/sdk-js-sveltekit. If you don't pay attention then you will be wondering why your code does not work. 

That is a biggy. Can we simulate the exports and see how the IDE behaves?

Reply:
![image](https://user-images.githubusercontent.com/21335119/226840699-095a4123-a2f0-4119-a2ff-0205f6bdcaae.png)

This is how it wlll look like. Sometimes the IDE shows the recently imported version at the top of the list and sometimes not..  There will be a lot of issues if we go with this approach.

Adding a custom path alias to `tsconfig.json` is also not a reliable option.
![image](https://user-images.githubusercontent.com/21335119/226841673-bfd3fde3-be82-46bd-b27e-7b299f5dc68b.png)
The "Quick fix..." panel will show it at the bottom, while "Code completion" shows it at the top
![image](https://user-images.githubusercontent.com/21335119/226842068-d7dbd7ed-f239-46e8-9e2f-797b526c6bda.png)

Reply:
Wait. Do we even need to expose common functions like `changeLanguage` on a per-adapter level? 

If the adapters are only for the bundler and the SDK is "auto generated" based on the adapter, regardless of the framework everyone can import from `@inlang/sdk-js` or am i mistaken?

Reply:
Right, we could just not expose them publically and the `plugin` could replace the function calls with the correct one. Since the functionality of `changeLanguage` will differ depending on the configuration we will need to do that anyways.